lib/pure/hashes.nim

```diff
@@ -127,26 +127,26 @@ when defined(js):
   import std/jsbigints
   import std/private/jsutils
 
-  proc noBigInt(): bool =
-    asm """`result` = typeof BigInt == 'undefined'"""
-
   proc hiXorLoJs(a, b: JsBigInt): JsBigInt =
-    let prod = a * b
-    let mask = (big"1" shl big"64") - big"1"
+    let
+      prod = a * b
+      mask = big"0xffffffffffffffff" # mask = (big"1" shl big"64") - big"1"
     result = (prod shr big"64") xor (prod and mask)
 
   template hashWangYiJS(x: JsBigInt): Hash =
-    let P0 = (big"0xa0761d64" shl big"32") or big"0x78bd642f"
-    let P1 = (big"0xe7037ed1" shl big"32") or big"0xa0b428db"
-    let P58 = (big"0xeb44acca" shl big"32") or (big"0xb455d165" xor big"8")
-    let res = hiXorLoJs(hiXorLoJs(P0, x xor P1), P58)
-    let maxSafeInterger = (big"1" shl big"53") - big"1"
+    let
+      P0 = big"0xa0761d6478bd642f"
+      P1 = big"0xe7037ed1a0b428db"
+      P58 = big"0xeb44accab455d165" xor big"8"
```

Comment:
no idea whether js will guarantee inlining this so how about:
```suggestion
      P58 = big"eb44accab455d16d"
```

Reply:
Should those be `const`, they are static values anyway.

Reply:
But it is not consistent with C backend.

Reply:
we can test those equivalences in thashes.nim and add a comment pointing to those tests; IMO performance of hashes is more important. Unless there's some guarantee this gets inlined.